crates/pathfinder/src/sequencer.rs

```diff
@@ -137,61 +137,33 @@ where
-                    None => {}
-                    Some(level) if level >= Level::TRACE => {
-                        trace!("Retrying due to: {:?}", e);
-                    }
-                    Some(_) => {
-                        if e.is_timeout() {
-                            info!("Retrying due to a timeout.");
-                        } else if e.is_decode() {
-                            warn!("Retrying due to a deserialization error.");
-                        } else if e.is_status() {
-                            match e.status() {
-                                Some(code) if code.is_redirection() => {
-                                    warn!("Retrying due to a redirect error: {code}")
-                                }
-                                Some(code @ StatusCode::NOT_FOUND) => {
-                                    info!("Retrying due to: {code}.")
-                                }
-                                Some(StatusCode::TOO_MANY_REQUESTS) => {
-                                    info!("Retrying due to rate limiting.")
-                                }
-                                Some(code) if code.is_client_error() => {
-                                    warn!("Retrying due to: {code}.")
-                                }
-                                Some(StatusCode::INTERNAL_SERVER_ERROR) => {
-                                    unreachable!("All HTTP 500s should either be converted to StarkNet errors or result in a deserialization error.")
-                                }
-                                Some(
-                                    code @ (StatusCode::BAD_GATEWAY
-                                    | StatusCode::SERVICE_UNAVAILABLE
-                                    | StatusCode::GATEWAY_TIMEOUT),
-                                ) => info!("Retrying due to: {code}."),
-                                Some(code) if code.is_server_error() => warn!("Retrying due to: {code}."),
-                                Some(code) => warn!("Retrying due to unclassified status code: {code}."),
-                                None => unreachable!(),
-                            }
-                        } else if e.is_redirect() {
-                            warn!("Retrying due to a redirect error.")
-                        } else if e.is_connect() {
-                            info!("Retrying due to a connection error.")
-                        } else if e.is_body() {
-                            info!("Retrying due to an error sending the request's or receiving the reply's body.")
-                        } else {
-                            // We should not get here since the only remaining error category
-                            // is a builder error
-                            warn!("Retrying due to an unexpected error.")
+                if e.is_body() || e.is_connect() || e.is_timeout() {
+                    info!(error=%e, "Request failed, retrying:");
```

Comment:
I don't like repeating the string literal, however I couldn't use a `const MSG: &str = "Request failed, retrying";` because structured logging would add the const's name to the log, while I wanted it to be `Request failed, retrying: error=Some error`, instead of `MSG="Request failed, retrying" error=Some error`.

Reply:
```suggestion
                    info!(reason=%e, "Request failed, retrying");
```
I think we should name it reason rather than error. And remove the colon (the structure shouldn't form part of the message).

Reply:
why is there a colon in this string?